### PR TITLE
 fix signature of ShimModel::initialize 

### PR DESCRIPTION
**Pull request by bancer**

### CODE REVIEW

**Comment by bancer** [comment]
Model.initialize event dispatch produced the error below. So, I removed it.
```
Error: Argument 1 passed to ShimModel::initialize() must be of the type array, object given, called in /home/travis/build/dereuromark/cakephp/lib/Cake/Event/CakeEventManager.php on line 243
#0 /home/travis/build/dereuromark/cakephp/lib/Cake/Event/CakeEventManager.php(243): ShimModel->initialize(Object(CakeEvent))
#1 /home/travis/build/dereuromark/cakephp/app/Plugin/Shim/Model/ShimModel.php(40): CakeEventManager->dispatch(Object(CakeEvent))
#2 [internal function]: ShimModel->__construct(Array)
#3 /home/travis/build/dereuromark/cakephp/lib/Cake/Utility/ClassRegistry.php(169): ReflectionClass->newInstance(Array)
#4 /home/travis/build/dereuromark/cakephp/app/Plugin/Shim/Test/Case/Model/ShimModelTest.php(18): ClassRegistry::init('ShimAppModelPos...')
#5 /home/travis/.composer/vendor/phpunit/phpunit/PHPUnit/Framework/TestCase.php(835): ShimModelTest->setUp()
#6 /home/travis/.composer/vendor/phpunit/phpunit/PHPUnit/Framework/TestResult.php(648): PHPUnit_Framework_TestCase->runBare()
#7 /home/travis/.composer/vendor/phpunit/phpunit/PHPUnit/Framework/TestCase.php(783): PHPUnit_Framework_TestResult->run(Object(ShimModelTest))
#8 /home/travis/build/dereuromark/cakephp/lib/Cake/TestSuite/CakeTestCase.php(84): PHPUnit_Framework_TestCase->run(Object(PHPUnit_Framework_TestResult))
#9 /home/travis/.composer/vendor/phpunit/phpunit/PHPUnit/Framework/TestSuite.php(779): CakeTestCase->run(Object(PHPUnit_Framework_TestResult))
#10 /home/travis/.composer/vendor/phpunit/phpunit/PHPUnit/Framework/TestSuite.php(749): PHPUnit_Framework_TestSuite->runTest(Object(ShimModelTest), Object(PHPUnit_Framework_TestResult))
#11 /home/travis/.composer/vendor/phpunit/phpunit/PHPUnit/Framework/TestSuite.php(708): PHPUnit_Framework_TestSuite->run(Object(PHPUnit_Framework_TestResult), false, Array, Array, false)
#12 /home/travis/.composer/vendor/phpunit/phpunit/PHPUnit/TextUI/TestRunner.php(349): PHPUnit_Framework_TestSuite->run(Object(PHPUnit_Framework_TestResult), false, Array, Array, false)
#13 /home/travis/build/dereuromark/cakephp/lib/Cake/TestSuite/CakeTestRunner.php(67): PHPUnit_TextUI_TestRunner->doRun(Object(CakeTestSuite), Array)
#14 /home/travis/build/dereuromark/cakephp/lib/Cake/TestSuite/CakeTestSuiteCommand.php(98): CakeTestRunner->doRun(Object(CakeTestSuite), Array)
#15 /home/travis/build/dereuromark/cakephp/lib/Cake/Console/Command/TestShell.php(283): CakeTestSuiteCommand->run(Array)
#16 /home/travis/build/dereuromark/cakephp/lib/Cake/Console/Command/TestShell.php(268): TestShell->_run(Array, Array)
#17 /home/travis/build/dereuromark/cakephp/lib/Cake/Console/Shell.php(462): TestShell->main()
#18 /home/travis/build/dereuromark/cakephp/lib/Cake/Console/ShellDispatcher.php(219): Shell->runCommand('Shim', Array)
#19 /home/travis/build/dereuromark/cakephp/lib/Cake/Console/ShellDispatcher.php(66): ShellDispatcher->dispatch()
#20 /home/travis/build/dereuromark/cakephp/app/Console/cake.php(47): ShellDispatcher::run(Array)
#21 {main}
```

File: Model/ShimModel.php [hunk 1]
```diff
@@ -712,14 +717,26 @@ public function record($id, array $options = []) {
...
+	 * Initialize a table instance. Called after the constructor.
+	 *
+	 * You can use this method to define associations, attach behaviors
+	 * define validation and do any other initialization logic you need.
 	 *
-	 * Will have param `array $config` in 3.x.
+	 * ```
+	 *  public function initialize(array $config)
+	 *  {
+	 *      $this->belongsTo('Users');
```
Comment (dereuromark) [inline Model/ShimModel.php]: These examples are not helpful, since those methods dont exist in 3.x.

File: Model/ShimModel.php [hunk 2]
```diff
@@ -712,14 +717,26 @@ public function record($id, array $options = []) {
 		return $this->get($id, $options);
 	}
 
+
 	/**
-	 * Model initialization callback. To avoid overwriting constructor
+	 * Initialize a table instance. Called after the constructor.
```
Comment (dereuromark) [inline Model/ShimModel.php]: table != model. better to keep things simple here, less wrong text is better.

File: Model/ShimModel.php [hunk 3]
```diff
@@ -29,8 +29,13 @@ public function __construct($id = false, $table = null, $ds = null) {
 			$this->prefixOrderProperty();
 		}
 
-		$event = new CakeEvent('Model.initialize', $this, compact('id', 'table', 'ds'));
```
Comment (dereuromark) [inline Model/ShimModel.php]: If you want to be close to 3.x, then removing this doesnt sound like a good idea
see https://github.com/cakephp/cakephp/blob/master/src/ORM/Table.php#L293

Comment (bancer) [reply Model/ShimModel.php]: This event produced the error I posted above.

Comment (dereuromark) [reply Model/ShimModel.php]: What exactly did you use? The one so far worked.

File: Model/ShimModel.php (hunk 1 above)
Comment (bancer) [reply Model/ShimModel.php]: I will change to singular and backport `setPrimaryKey`.

File: Model/ShimModel.php (hunk 3 above)
Comment (bancer) [reply Model/ShimModel.php]: https://travis-ci.org/dereuromark/cakephp-shim/jobs/366295370

File: Model/ShimModel.php (hunk 1 above)
Comment (dereuromark) [reply Model/ShimModel.php]: I wouldnt backport the new 3.6 setter ones IMO. too many new methods.
Lets keep things simple.

File: Model/ShimModel.php (hunk 3 above)
Comment (dereuromark) [reply Model/ShimModel.php]: ok

**Pull request merged**